Added: Enable Download Client Priorities

Qstick opened the pull request:
#### Database Migration
NO

#### Description

Re-enable Download Client Priorities

#### Issues Fixed or Closed by this PR

Fixes #229

galli-leo commented on src/NzbDrone.Core/Parser/Model/RemoteMovie.cs:
```diff
@@ -12,6 +12,16 @@ public class RemoteMovie
         public Movie Movie { get; set; }
         public MappingResultType MappingResult { get; set; }
 
+        public bool IsRecentMovie()
```
IMO this makes much more sense being on the actual movie object.

Qstick replied:
I kinda agree, I just added where Sonarr had it.

galli-leo replied:
Ah interesting, I would still move it to the movie itself (and also make it a property). Maybe we need it somewhere else in the future.

galli-leo commented on src/UI/Form/CheckboxTemplate.hbs:
```diff
@@ -4,7 +4,7 @@
         <div class="input-group">
             <label class="checkbox toggle well">
-                <input type="checkbox" name="fields.{{order}}.value"/>
+                <input type="checkbox" name="fields.{{order}}.value" validation-name="{{name}}"/>
```
Why did you add this?

Qstick commented:
@galli-leo, I moved the method over to Movie.cs, however I don't know that it makes since for this to be a property instead of a method, we would have to constantly recalculate it as it would change daily. Seems to make more sense to leave as a method instead. 

Additionally, I noticed many of my older movies only had a cinema date and not a physical release so I added a backup to the logic. If physical release is in the last 21 days then isRecent, if cinema date is in the last 120 days then isRecent. If neither Physical or Cinema have a date (I don't know that this is even possible, but good to have backup case) then isRecent. Tested on a few nzbs and priority is set as intended.

danjames92 commented:
Wouldn't it make more sense to fix the bug with the priorities first and then make any changes to the priorities later in a separate PR?

I'm not sure why the behaviour needs to the change as such.

galli-leo commented:
@Qstick I think the 120 days from cinemas is a good option.

I still think it should be a property. Don’t really understand how we would have to recalculate it daily? Also isAvailable works the same and since it’s just data, a property makes more sense (imo).

I know, I am being pedantic, but release is gonna have to wait anyways 😅

Qstick commented:
@galli-leo is available is setup same as I have this setup. 

https://github.com/Radarr/Radarr/blob/25e10e26e308b4ad41dcaa428e4a171413b2ae08/src/NzbDrone.Core/Movies/Movie.cs#L77

It's a method for the movie model, but a property for the API resource that calls back to the model method. Is that what you mean?

galli-leo commented:
@Qstick Ah yeah forgot about the delay. But yeah I would make isRecent a property like in the API model, since we don’t need a parameter like with isAvailable.
